### DISCUSSION TITLE
Finetune Text-to-Speech: Tacotron2 and HifiGAN

### DISCUSSION THREAD

**Discussion by olahsymbo100** [speechbrain/speechbrain#1676] [Q&A]
Hi,

I have been able to set up speechbrain on my linux server. And I am currently training the Text-to-Speech Tacotron2 and HifiGAN. However, I would really like to use transfer learning instead of training from scratch. I’ve searched the entire documentation but cant find how to finetune either Tacotron2 or HifiGAN to smaller dataset. Please guide me on this/point me to existing implementation for this. I would really appreciate your help.

Thank you.
Kind regards.

**Comment by BenoitWang** [comment]
Hi @olahsymbo100 thank you for your interest. We don't have a TTS fine-tuning tuto. But if you want to fine-tune a TTS model, you can load the model and fine-tune it with the `pretrainer` [class](https://colab.research.google.com/drive/1LN7R3U3xneDgDRK2gC5MzGkLysCWxuC3#scrollTo=l7qb4lrtG8G-
). 

And if you use an English fine-tuning dataset, you only need to prepare train/valid json files with the same format as LJSpeech. But if you want to fine-tune the model to other languages, you'll need a new text pipeline as in `speechbrain.utils.text_to_sequence`.

**Comment by olahsymbo100** [comment]
Hi @BenoitWang Thank you very much for your response. I have been able to implement the pretrainer class using the link you shared and its able to load the pretrained model. However, I'm a bit confuse on how to proceed with finetuning (fit) the pretrained model to my dataset. I've checked the PreTrainer class it doesnt include any `fit` function and neither does it have a routine to pass the loaded model to an existing training class. 

In other words, PreTrainer class just fetches and loads the model, there is no way to continue training. 
Could you please clarify.

Thank you

**Comment by BenoitWang** [comment]
Hi @olahsymbo100, the `pretrainer` class does not have a fit method. Once the `pretrainer` loads the modules, you only need to launch the same training pipeline as in `train.py` with your own data. You can read the entire tuto in the same link.

**Comment by olahsymbo100** [comment]
Hi @BenoitWang thank you for the clarification. I'm currently implementing it as:

```
    datasets = dataio_prepare(hparams)

    # Brain class initialization
    tacotron2_brain = Tacotron2Brain(
        modules=hparams["modules"],
        opt_class=hparams["opt_class"],
        hparams=hparams,
        run_opts=run_opts,
        checkpointer=hparams["checkpointer"],
    )

    # Initialization of the pre-trainer
    pretrain = Pretrainer(collect_in='model_local',
                          loadables={'model': Tacotron2.Tacotron2()},
                          paths={'model': 'speechbrain/tts-tacotron2-ljspeech/model.ckpt'})

    # We download the pretrained model from HuggingFace in this case
    pretrain.collect_files()
    pretrain.load_collected(device='cpu')

    tacotron2_brain.fit(
        tacotron2_brain.hparams.epoch_counter,
        train_set=datasets["train"],
        valid_set=datasets["valid"],
        train_loader_kwargs=hparams["train_dataloader_opts"],
        valid_loader_kwargs=hparams["valid_dataloader_opts"],
    )

```

I'm just wondering if ` checkpointer=hparams["checkpointer"]` will not make it override the loaded pretrained model.
